packages/fast-check/src/arbitrary/_internals/helpers/DepthContext.ts

```diff
@@ -11,7 +11,10 @@ declare const depthIdentifierSymbol: unique symbol;
  * @remarks Since 2.25.0
  * @public
  */
-export type DepthIdentifier = { [depthIdentifierSymbol]: true };
+export type DepthIdentifier = {
```

Comment:
With the suggested approach users will be able to replace instances of `DepthIdentifier` by raw `{}`. While I'm ok with relaxing the constraint for now, I'd rather put: `export type DepthIdentifier = {/* the symbol thing */} & DepthContext`.

Here is a repro of the problem: https://github.com/dubzzz/fast-check-type-hazard-repro/blob/main/main.ts#L14

Line 14 compiles with your suggestion. Ideally I'd at least force the users to come with identifiers matching context type. In other words, for end-users it will result into Identifier type is just Context (with another name), but I'm totally ok with that (until dual goes away).